worldengine/draw.py

```diff
@@ -129,6 +210,109 @@ def elevation_color(elevation, sea_level=1.0):
     return _sature_color(_elevation_color(elevation, sea_level))
 
 
+def get_normalized_elevation_mask(world):
+    ''' Convert raw elevation into normalized values between 0 and 255,
+        and return a numpy arrawy of these values '''
+
+    e = world.elevation['data']
+
+    mask = numpy.ma.array(e, mask=world.ocean)  # only land
+    min_elev_land = mask.min()
+    max_elev_land = mask.max()
+    elev_delta_land = max_elev_land - min_elev_land
+
+    mask = numpy.ma.array(e, mask=numpy.logical_not(world.ocean))  # only ocean
+    min_elev_sea = mask.min()
+    max_elev_sea = mask.max()
+    elev_delta_sea = max_elev_sea - min_elev_sea
+
+    c = numpy.empty(e.shape, dtype=numpy.float)
+    c[numpy.invert(world.ocean)] = (e[numpy.invert(world.ocean)] - min_elev_land) * 127 / elev_delta_land + 128
+    c[world.ocean] = (e[world.ocean] - min_elev_sea) * 127 / elev_delta_sea
+    c = numpy.rint(c).astype(dtype=numpy.int32)  # proper rounding
+
+    return c
+
+
+def get_biome_color_based_on_elevation(world, elev, x, y):
+    ''' This is the "business logic" for determining the base biome color in satellite view.
+        This includes generating some "noise" at each spot in a pixel's rgb value, potentially 
+        modifying the noise based on elevation, and finally incorporating this with the base biome color. 
+
+        The basic rules regarding noise generation are:
+        - Oceans have no noise added
+        - land tiles start with noise somewhere inside (-NOISE_RANGE, NOISE_RANGE) for each rgb value
+        - land tiles with high elevations further modify the noise by set amounts (to drain some of the 
+          color and make the map look more like mountains) 
+
+        The biome's base color may be interpolated with a predefined mountain brown color if the elevation is high enough.
+
+        Finally, the noise plus the biome color are added and returned
+    '''
+    v = world.biome[y, x]
+    biome_r, biome_g, biome_b = _biome_satellite_colors[v]
+
+    # Default is no noise - will be overwritten if this tile is land
+    noise = (0, 0, 0)
+
+    if world.is_land((x, y)):
```

Comment:
As suggested by @esampson, don't forget about the else case. In there you could check the temperature map and 'freeze' the ocean if necessary.